### PR TITLE
Add lychee github action  for links checking

### PR DESCRIPTION
**Pull request by spapadop**
### Description
Adds a links checker to ensure that all links used in the repo are valid.

### Issues Resolved
closes https://github.com/opensearch-project/opensearch-api-specification/issues/268

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nhtruong** [comment]
Great work thank you.
Note that you have to add a verb like `close #268`  in the PR description for issue #268 to be closed automatically when this PR is merged. (I've just updated the description accordingly)

File: .github/workflows/links-fail-fast.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@v1
+        with:
+          fail: true
+          args: --base . --verbose --no-progress './**/*.yaml' './**/*.yml' './**/*.md'
```
Comment (nhtruong) [inline .github/workflows/links-fail-fast.yml:17]: we should include `.json` and `.ts` files too.

File: .github/workflows/links-fail-fast.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Links (Fail Fast)
```
Comment (dblock) [inline .github/workflows/links-fail-fast.yml:1]: What does "(Fail Fast)" mean in this context? Why isn't it just links?

Comment (spapadop) [reply .github/workflows/links-fail-fast.yml]: Renamed to links

**Review by dblock — APPROVED** [review]

**Review by dblock — CHANGES_REQUESTED** [review]
I see links from npmjs checked.

```
✔ [200] https://registry.npmjs.org/pretty-format/-/pretty-format-29.7.0.tgz
✔ [200] https://registry.npmjs.org/@types/istanbul-reports/-/istanbul-reports-3.0.4.tgz
```

Exclude package-lock.json, possibly others.

**Review by dblock — APPROVED** [review]
Thanks for contributing this! Hope you can pickup something else, lmk if you want suggestions :)

**Pull request merged**